src/transaction-runner.ts

```diff
@@ -233,8 +235,18 @@ export abstract class Runner<T> {
       const transaction = await this.getTransaction();
 
       try {
-        return await this._run(transaction);
+        const result = await this._run(transaction);
+        if (this.attempts > 0) {
```

Comment:
+1 exists to make it human readable because `attempt=0` is great for programmers but not intuitive for people counting. Sure I'll remove that if condition.

Reply:
this.attempts refers to attempts after retrying and not the loop iterations hence for it to make sense to a human being, the +1 is necessary otherwise they'll be very confused about the number of attempts.